crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -166,8 +166,13 @@ impl Solution {
         let access_list = simulator.access_list(tx.clone()).await?;
         let tx = tx.set_access_list(access_list.clone());
 
-        // Finally, get the gas for the settlement using the full access list.
-        let gas = simulator.gas(tx).await?;
+        // Third, get the gas parameters for the settlement using the full
+        // access list.
+        let gas = {
+            let estimate = simulator.gas(tx).await?;
+            let price = eth.gas_price().await?;
```

Comment:
`try_join!` doesn't do automatic error conversions with `From` 😮.

Reply:
I left this out for now for two reasons:
1. It made the code uglier (subjective)
2. It doesn't have any real effect because of how `NativeGasEstimator` works under the hood (it updates it internal values in a background task, so reading the estimate isn't really blocking).

Reply:
> it updates it internal values in a background task, so reading the estimate isn't really async

Didn't know that. Then I agree that we should keep the nicer looking code.